buildSrc/src/main/java/quilt/internal/MappingsExtension.java

```diff
@@ -1,15 +1,54 @@
 package quilt.internal;
 
+import org.gradle.api.GradleException;
 import org.gradle.api.Project;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Provider;
+import org.quiltmc.enigma.api.EnigmaProfile;
+
+import java.io.File;
+import java.io.IOException;
+
+public abstract class MappingsExtension {
+    public final Provider<EnigmaProfile> enigmaProfile;
 
-public class MappingsExtension {
     private final FileConstants fileConstants;
 
-    public MappingsExtension(Project target) {
-        fileConstants = new FileConstants(target);
+    public static MappingsExtension get(Project project) {
+        return project.getExtensions().getByType(MappingsExtension.class);
+    }
+
+    protected abstract RegularFileProperty getEnigmaProfileFile();
+
+    public MappingsExtension(Project project) {
+        this.fileConstants = new FileConstants(project);
+
+        this.enigmaProfile = this.getEnigmaProfileFile()
```

Comment:
`map`ed providers _are not cached_, and I assume this is used in several placed  -- so instead of having, say, a `public final Provider<EnigmaProfile> enigmaProfile`, have a `Property<EnigmaProfile>` and `set` it to the mapped provider, which _is_ cached